live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/request/ATSTransformRequester.java

```diff
@@ -96,9 +104,9 @@ private Map<String, String> getTransformRequestOptions(Transform transformProper
         return Map.of(TIMEOUT_KEY, String.valueOf(transformProperties.request().timeout()));
     }
 
-    private String makeClientDataString(TransformRequest transformRequest)
+    private String makeClientDataString(TransformRequest transformRequest, int attempt)
     {
-        ClientData clientData = new ClientData(transformRequest.nodeRef(), transformRequest.targetMimeType());
+        ClientData clientData = new ClientData(transformRequest.nodeRef(), transformRequest.targetMimeType(), attempt);
         try
         {
             return objectMapper.writeValueAsString(clientData);
```

Comment:
Is it required here, after you added `ClientDataSerializer`? 🤔

Reply:
I actually can use it - I refactored this class a little

Reply:
Nice refactor!